packages/client/src/MapLayerToggle.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useMemo } from "react";
```

Comment:
I think it makes sense to move some of this into a `react` directory to distinguish it from the other parts of the stack (phaser, noa, etc.) and drop some of the UI things into a `ui` directory or something. I was planning to take an organizational pass at the very end once things became more clear, can do in this PR or as a follow up.

I should note that I'm not a huge fan of the `components` directory pattern that you see in a lot of example apps/repos. I tend to structure things so that functional parts of the app are together, e.g. a set of components and hooks and utils/modules related to the map view are together in a `map` directory. And I tend to wait on organizing until it feels necessary rather than doing it too soon and having only one file in a directory.

Reply:
here's what I ended up with:
<img width="272" alt="image" src="https://user-images.githubusercontent.com/508855/203701359-ef723acf-529d-4272-94da-005ae14f5772.png">

~everything related to showing UI over the app layers (phaser/noa) is in `ui`